apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/seller-celebration-modal/index.jsx

```diff
@@ -0,0 +1,130 @@
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const [ hasDisplayedModal, setHasDisplayedModal ] = useState( false );
+	const isSiteEditor = useSelect( ( select ) => !! select( 'core/edit-site' ) );
+	const previousIsEditorSaving = useRef( false );
+	const { isEditorSaving, hasPaymentsBlock, linkUrl } = useSelect( ( select ) => {
+		if ( isSiteEditor ) {
+			const isSavingSite = select( 'core' ).isSavingEntityRecord( 'root', 'site' );
+			const page = select( 'core/edit-site' ).getPage();
+			const pageId = parseInt( page?.context?.postId );
+			const isSavingEntity = select( 'core' ).isSavingEntityRecord( 'postType', 'page', pageId );
+			const pageEntity = select( 'core' ).getEntityRecord( 'postType', 'page', pageId );
+			const rawContent = pageEntity.content.raw;
+			return {
+				isEditorSaving: isSavingSite || isSavingEntity,
+				hasPaymentsBlock: rawContent.includes( '<!-- wp:jetpack/recurring-payments -->' ),
+				linkUrl: pageEntity.link,
+			};
+		}
+		const currentPost = select( 'core/editor' ).getCurrentPost();
+		const isSavingEntity = select( 'core' ).isSavingEntityRecord(
+			'postType',
+			currentPost?.type,
+			currentPost?.id
+		);
+		const globalBlockCount = select( 'core/block-editor' ).getGlobalBlockCount(
+			'jetpack/recurring-payments'
+		);
+		return {
+			isEditorSaving: isSavingEntity,
+			hasPaymentsBlock: globalBlockCount > 0,
+			linkUrl: currentPost.link,
+		};
+	} );
+	const intent = useSiteIntent();
+	const {
+		hasSeenSellerCelebrationModal,
+		updateHasSeenSellerCelebrationModal,
+	} = useHasSeenSellerCelebrationModal();
+
+	useEffect( () => {
+		if (
+			! isEditorSaving &&
+			previousIsEditorSaving.current &&
+			! hasDisplayedModal &&
+			intent === 'sell' &&
+			hasPaymentsBlock &&
+			! hasSeenSellerCelebrationModal
+		) {
+			setIsModalOpen( true );
+			setHasDisplayedModal( true );
+			updateHasSeenSellerCelebrationModal( true );
+		}
+		previousIsEditorSaving.current = isEditorSaving;
+	}, [
+		isEditorSaving,
+		hasDisplayedModal,
+		intent,
+		hasPaymentsBlock,
+		hasSeenSellerCelebrationModal,
+		updateHasSeenSellerCelebrationModal,
+	] );
+
+	// if save state has changed and was saving on last render
+	// then it has finished saving
+	// open modal if content has sell block,
+
+	const closeModal = () => setIsModalOpen( false );
+	return (
+		<NuxModal
+			isOpen={ isModalOpen }
+			className="wpcom-site-editor-seller-celebration-modal"
+			title={ __( "You've added your first product!", 'full-site-editing' ) }
+			description={ __(
+				'Preview your product on your site before launching and sharing with others.',
+				'full-site-editing'
+			) }
+			imageSrc={ contentSubmittedImage }
+			actionButtons={
+				<>
+					<Button onClick={ closeModal }>{ __( 'Continue Editing', 'full-site-editing' ) }</Button>
+					<Button isPrimary href={ linkUrl } target="__blank" rel="noopener noreferrer">
+						{ __( 'View Post', 'full-site-editing' ) }
```

Comment:
Super tiny nitpick, but IIRC we're supposed to use sentence case for buttons/actions (so "View post" and "Continue editing"). I don't know if that's the same in the editor, but we made an effort to do so in Calypso: https://github.com/Automattic/wp-calypso/blob/trunk/client/components/forms/README.md (see last sentence)

Reply:
Aha, good catch. (I actually just realized that "View Post" is the wrong text per the new mockups anyway, so I will fix this.)